stac_fastapi/pgstac/stac_fastapi/pgstac/config.py

```diff
@@ -0,0 +1,39 @@
+"""Postgres API configuration."""
+from typing import Set
+
+from stac_fastapi.types.config import ApiSettings
+
+
+class Settings(ApiSettings):
+    """Postgres-specific API settings.
+
+    Attributes:
+        postgres_user: postgres username.
+        postgres_pass: postgres password.
+        postgres_host_reader: hostname for the reader connection.
+        postgres_host_writer: hostname for the writer connection.
+        postgres_port: database port.
+        postgres_dbname: database name.
+    """
+
+    postgres_user: str
+    postgres_pass: str
+    postgres_host_reader: str
+    postgres_host_writer: str
+    postgres_port: str
+    postgres_dbname: str
+
+    db_min_conn_size: int = 10
+    db_max_conn_size: int = 10
+    db_max_queries: int = 50000
+    db_max_inactive_conn_lifetime: float = 300
+
+    @property
+    def reader_connection_string(self):
+        """Create reader psql connection string."""
+        return f"postgresql://{self.postgres_user}:{self.postgres_pass}@{self.postgres_host_reader}:{self.postgres_port}/{self.postgres_dbname}"
```

Comment:
I see pydantic has a [`PostgresDsn` type](https://pydantic-docs.helpmanual.io/usage/types/#urls). Maybe it would be useful for validation purposes to have `reader_connection_string` be a field with `init=False` that's set within `__post_init__`? Then you'd have fuller validation?

Reply:
We can look at that later (across both pgstac and sqlalchemy - which this is set up identically to right now). I know I ran into some issue initially when I did use the PostgresDsn type, I think it had some issues when you didn't directly supply some non-required fields IIRC.